pkg/controllers/state/suite_test.go

```diff
@@ -99,6 +99,26 @@ var _ = AfterEach(func() {
 	cluster.Reset()
 	cloudProvider.Reset()
 })
+var _ = Describe("Pod Healthy NodePool", func() {
+	It("should not store pod schedulable time if the nodePool that pod is scheduled to does not have NodeRegistrationHealthy=true", func() {
+		pod := test.Pod()
+		ExpectApplied(ctx, env.Client, pod, nodePool)
+
+		cluster.UpdatePodHealthyNodePoolScheduledTime(ctx, nodePool.Name, time.Now(), []*corev1.Pod{pod}...)
+		setTime := cluster.PodSchedulingSuccessTimeRegistrationHealthyCheck(client.ObjectKeyFromObject(pod))
+		Expect(setTime.IsZero()).To(BeTrue())
+	})
+	It("should store pod schedulable time if the nodePool that pod is scheduled to has NodeRegistrationHealthy=true", func() {
```

Comment:
I think we need another test that ensures we don't overwrite the time if it's already stored AND that we properly clean things up when the pod is deleted